coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java

```diff
@@ -0,0 +1,330 @@
+    }
+
+    SqlNode join = new SqlJoin(POS, leftResult.asFrom(), SqlLiteral.createBoolean(false, POS),
+        JoinType.COMMA.symbol(POS), rightSqlNode, JoinConditionType.NONE.symbol(POS), null);
+
+    return result(join, leftResult, rightResult);
+  }
+
+  /**
+   * Custom table-valued functions are represented as LogicalTableFunctionScan type relational expression.
+   * Current version of Calcite used in Coral does not implement traversing
+   * a LogicalTableFunctionScan type RelNode. Hence, this implementation is added.
+   *
+   * @param e RelNode of type LogicalTableFunctionScan as input. Example:
+   *           <pre>
+   *             LogicalTableFunctionScan(invocation=[foo_udtf_CountOfRow($cor0.a)])
+   *           </pre>
+   *
+   * @return Result of converting the RelNode to a SqlNode.
+   *         The conversion result of the above example is:
+   *         <pre>
+   *                                (SqlBasicCall)
+   *          COLLECTION_TABLE(`foo_udtf_CountOfRow`(`complex`.`a`))
+   *                                     |
+   *            _________________________|__________________________
+   *           |                                                   |
+   *  Operator: COLLECTION_TABLE               Operand: `foo_udtf_CountOfRow`(`complex`.`a`)
+   *                                                              |
+   *                                             _________________|______________________
+   *                                            |                                       |
+   *                             Operator: foo_udtf_CountOfRow               Operand: `complex`.`a`
+   *         </pre>
+   */
+  public Result visit(LogicalTableFunctionScan e) {
+    RexCall call = (RexCall) e.getCall();
+    SqlOperator functionOperator = call.getOperator();
+    final List<SqlNode> functionOperands = new ArrayList<>();
+    for (RexNode rexOperand : call.getOperands()) {
+      RexFieldAccess rexFieldAccess = (RexFieldAccess) rexOperand;
+      RexCorrelVariable rexCorrelVariable = (RexCorrelVariable) rexFieldAccess.getReferenceExpr();
+      SqlNode sqlNodeOperand = correlTableMap.get(rexCorrelVariable.id).toSql(null, rexOperand);
+      functionOperands.add(sqlNodeOperand);
+    }
+
+    SqlCall functionSqlCall = functionOperator.createCall(POS, functionOperands.toArray(new SqlNode[0]));
+
+    SqlNode tableCall = new SqlLateralOperator(SqlKind.COLLECTION_TABLE).createCall(POS, functionSqlCall);
+
+    Result tableCallResultWithAlias = result(tableCall, ImmutableList.of(Clause.FROM), e, null);
+
+    return new Result(tableCall, ImmutableList.of(Clause.FROM), null, e.getRowType(),
+        ImmutableMap.of(tableCallResultWithAlias.neededAlias, e.getRowType()));
+  }
+
+  /**
+   * Join RelNode represents a Node with two child relational expressions linked by different join type conditions.
+   * When the right child of a Correlate node is an Uncollect / TableFunction type RelNode,
```

Comment:
Is the reference to `Correlate` here accurate?

Reply:
thanks for pointing it out. That was a typo